### PR TITLE
`pip` install dask on gpuCI builds

### PR DESCRIPTION
**Pull request by charlesbluca**
Should resolve the failures observed in https://github.com/dask/dask/pull/9809#pullrequestreview-1241243229

cc @jrbourbeau 

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: continuous_integration/gpuci/build.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ gpuci_logger "Install distributed"
 python -m pip install git+https://github.com/dask/distributed
 
 gpuci_logger "Install dask"
-python setup.py install
+python -m pip install --quiet --no-deps -e .
```
Comment (jrbourbeau) [inline continuous_integration/gpuci/build.sh:44]: Not sure how noisy this is, but I can imagine wanting this output in the future for debugging (just a hunch) 

```suggestion
python -m pip install --no-deps -e .
```

Comment (jrbourbeau) [reply continuous_integration/gpuci/build.sh:44]: Why can we use `--no-deps`? Are those being installed in the base image or somewhere else?

**Review by jrbourbeau — COMMENTED** [review]
Thanks @charlesbluca! Just a couple of minor comments -- overall this looks like a nice improvement

File: continuous_integration/gpuci/build.sh (hunk 1 above)
Comment (charlesbluca) [reply continuous_integration/gpuci/build.sh:44]: Yeah, the gpuCI conda environment is the Dask python 3.9 CI environment with a few RAPIDS packages installed on top ([Dockerfile](https://github.com/rapidsai/dask-build-environment/blob/dfa14787ca32e96cb22c3c46352f4bfa1d0536c2/dask/Dockerfile#L13) for reference).

**Review by jrbourbeau — APPROVED** [review]
Thanks @charlesbluca! This is in

**Pull request merged**